### PR TITLE
Corrected behavior of SettingsValidate. Fixes #62

### LINKED ISSUE #62: Manually changing the settings (to nonsense) can prevent the user from opening his database
**Steps to reproduce this bug:**

- Change the TOTP settings of an entry manually to: ``0;5`` (both values are not allowed)
- Reopen database.
- ``System.DivideByZeroException`` will be thrown and the KeePass GUI won't be loaded correctly:

grafik

**The problem:**

The ``SettingsValidate`` method even returns true if the settings are not valid.

### PR DESCRIPTION
**Pull request by dannoe**
Fixes #62

### CODE REVIEW

**Comment by dannoe** [comment]
Changed the pr.yml file to not build the project on test. It is already build before by msbuild.

It looks like you can not build net45 projects with ``dotnet build`` anymore. see https://github.com/microsoft/msbuild/issues/4704

File: KeeTrayTOTP/TrayTOTP_Plugin.cs [hunk 1]
```diff
@@ -74,7 +74,7 @@ public sealed partial class KeeTrayTOTPExt : Plugin
         /// </summary>
         internal const int setstat_int_EntryList_RefreshRate = 300;
         internal const int setstat_trim_text_length = 25;
-        public IReadOnlyList<string> setstat_allowed_lengths = new ReadOnlyCollection<string>(new[] { "6", "7", "8", "S" });
+        internal IReadOnlyList<string> setstat_allowed_lengths = new ReadOnlyCollection<string>(new[] { "6", "7", "8", "S" });
```
Comment (robinvanpoppel) [inline KeeTrayTOTP/TrayTOTP_Plugin.cs]: Just a suggestion: This one could be private, but I see that the visibility in the rest of the files is mostly internal. I have some plans to refactor that when I have some more time on my hands.

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -22,4 +22,5 @@ jobs:
       run: msbuild KeeTrayTOTP.sln
 
     - name: Run tests
-      run: dotnet test
+      # It looks like the dotnet CLI can no longer build net45 projects? see https://github.com/microsoft/msbuild/issues/4704
+      run: dotnet test --no-build
```
Comment (robinvanpoppel) [inline .github/workflows/pr.yml]: Thanks for fixing this, as well as referencing the issue :)

File: KeeTrayTOTP/TrayTOTP_Plugin.cs [hunk 3]
```diff
@@ -702,45 +703,60 @@ internal bool SettingsValidate(PwEntry pe, out bool IsUrlValid)
         /// <returns>Error(s) while validating Interval or Length.</returns>
         internal bool SettingsValidate(PwEntry pe, out bool IsIntervalValid, out bool IsLengthValid, out bool IsUrlValid)
         {
-            bool SettingsValid = true;
+            bool settingsValid = true;
```
Comment (robinvanpoppel) [inline KeeTrayTOTP/TrayTOTP_Plugin.cs]: This makes it a lot more readable already. 👍

File: KeeTrayTOTP/TrayTOTP_Plugin.cs (hunk 1 above)
Comment (dannoe) [reply KeeTrayTOTP/TrayTOTP_Plugin.cs]: I would even suggest an enum for this, but I didn't want to make too much changes in one PR.

**Pull request merged**